docs/feeds.md

```diff
@@ -0,0 +1,106 @@
+
+The feed action named *changes* takes these parameters, and is expected to take whatever action is necessary to set up a stream of events from Cloudant, with the appropriate configuration, directed to the trigger *T*.    
+
+For the Cloudant *changes* feed, the action happens to talk directly to a *cloudant trigger* service we've implemented with a connection-based architecture.   We'll discuss the other architectures below.
+
+A similar feed action protocol occurs for `wsk trigger delete`.    
+
+We have not yet implemented pause and unpause, but they will be similar.
+
+# Implementing Feeds with Hooks
+
+Setting up a feed via a hook is *by far* the easiest way to prototype a new feed,
+if the even producer supports a webhook
+facility.
+
+
+With this method there is _no need_ to stand up any persistent service outside of OpenWhisk.  All feed management happens naturally though stateless OpenWhisk *feed actions*, which negotiate
+directly with a third part webhook API.
+
+When invoked with `CREATE`, the feed action simply installs a webhook for some other service, asking the remote service to POST notifications to the appropriate `fireTrigger` URL in OpenWhisk.
+
+The webhook should be directed to send notifications to a URL such as:
+`POST /namespaces/{namespace}/triggers/{triggerName}`
+
+# Implementing Feeds with Polling
+
+It is possible to set up an OpenWhisk *action* to poll a feed source entirely within OpenWhisk, without the need to stand up any persistent connections or external service.
+
+For feeds where a webhook is not available, but do not need high-volume or low latency response times, polling is an attractive option.
+
+To set up a polling-based feed, the feed action takes the following steps when called for `CREATE`:
+1.   The feed action sets up a periodic trigger (*T*) with the desired frequency, using the `whisk.system/alarms` feed.
+2.   The feed developer creates a `pollMyService' action which simply polls the remote service and returns any new events.
+3.  The feed action sets up a *rule*
+*T -> pollMyService*.
+
+That's it.   This procedure implements a polling-based trigger entirely using OpenWhisk actions, without any need for a separate service.
+
+# Implementing Feeds via Connections
+
+The previous 2 architectural choices are simple and easy to implement. However, if you want a high-performance feed, there is no substitute for persistent connections and long-polling or similar techniques.
+
+Since OpenWhisk actions are stateless,  an action cannot maintain a persistent connection to a third party. Instead, we are forced to stand up a separate service (outside of OpenWhisk) that runs all the time.   We call these *provider services*.  A provider service can maintain connections to third party event sources that support long polling or other connection-based notifications.   
```

Comment:
Since OpenWhisk actions are stateless,  an action cannot maintain a persistent connection to a third party.
I would say that cannot a persistent connection not because they are stateless but because they have a limit on how long they can run.
Maybe:
Since OpenWhisk actions are designed to be short lived they can't maintain a persistent connection to a third party service. To implement a feed based on long running process you would required to use a service outside of OpenWhisk that runs all the time. We call these *event provider* services.